datasets/us/doh_debarred/crawler.py

```diff
@@ -0,0 +1,51 @@
+from zavod import Context
+import csv
+import re
+
+def parsedate(date: str):
+    """Parse date from string."""
+    if date:
+        return f'{date[:4]}-{date[4:6]}-{date[6:]}'
+    return None
+
+def getdata(context: Context):
+    """Process data from CSV file."""
+    source_file = context.fetch_resource("source.csv", context.data_url)
+
+    with open(source_file, "r") as f:
+        reader = csv.DictReader(f)
+        for row in reader:
+            if row["LASTNAME"].strip() or row["FIRSTNAME"].strip():
+                entity = context.make("Person")
+                firstname = row["FIRSTNAME"]
+                midname = row["MIDNAME"]
+                lastname = row["LASTNAME"]
+                name = f"{firstname} {midname} {lastname}".strip()
+                entity.id = context.make_id(name)
+                entity.add("name", name)
+                entity.add("notes", row.get("GENERAL"))
+                entity.add("birthDate", parsedate(row.get("DOB")))
+                entity.add("address", row.get("ADDRESS"))
+                entity.add("createdAt", parsedate(row.get("EXCLDATE")))
+                entity.add("topics", "debarment")
+                entity.add("country", "us")
+                context.emit(entity, target=True)
+            else:
+                comp = context.make("Company")
+                address = row.get('ADDRESS', '').strip()
+                if address:
+                    match = re.match(r'^([#\w\s/-]+)', address)
+                    first_line = match.group(1).strip() if match else address
```

Comment:
you can just use the address column as it is - no need to regex

Reply:
ok. Also, there were some cases where the address row was empty (we are using address as id, so it can't be empty), In that case, what could be the solution?